File: ERP/erp-backend/src/main/java/com/liziczh/erp/service/InfoService.java
package com.liziczh.erp.service;

import com.liziczh.erp.entity.*;
import org.springframework.data.domain.Page;

import java.util.List;

/**
 * @Description 基础资料服务
 * @Author czh
 * @Date 2019/4/28
 */

public interface InfoService {

    /**************************************供应商*********************************************/
    /**
     * 分页查询供应商
     * @param typeId
     * @param search
     * @param currentPage
     * @param limit
     * @return
     */
    Page<Supplier> pagingQuerySupplierList(Integer typeId, String search, Integer currentPage, Integer limit);

    /**
     * 查询所有供应商
     *
     * @return List<Supplier>
     */
    List<Supplier> findAllSupplier();

    /**
     * 查询所有供应商分类
     * @return
     */
    List<SupplierType> findAllSupplierType();

    /**
     * 根据供应商ID查询单个供应商
     *
     * @param supplierId
     * @return Supplier
     */
    Supplier findSupplierBySupplierId(long supplierId);

    /**
     * 新增供应商信息
     *
     * @param supplier
     * @return Supplier
     */
    Supplier addSupplier(Supplier supplier);

    /**
     * 保存供应商信息
     *
     * @param supplier
     * @return Supplier
     */
    Supplier modifySupplier(Supplier supplier);

    /**
     * 移除供应商
     *
     * @param supplierId
     */
    void removeSupplier(long supplierId);


    /**************************************客户*********************************************/

    /**
     * 分页查找客户
     * @param typeId
     * @param search
     * @param currentPage
     * @param limit
     * @return
     */
    Page<Customer> pagingQueryCustomerList(Integer typeId, String search, Integer currentPage, Integer limit);

    /**
     * 查询所有客户
     *
     * @return List<Customer>
     */
    List<Customer> findAllCustomer();

    /**
     * 查询所有客户分类
     * @return
     */
    List<CustomerType> findAllCustomerType();

    /**
     * 根据客户ID查询单个客户
     *
     * @param customerId
     * @return Customer
     */
    Customer findCustomerByCustomerId(long customerId);

    /**
     * 添加客户信息
     *
     * @param customer
     * @return Customer
     */
    Customer addCustomer(Customer customer);

    /**
     * 修改客户信息
     *
     * @param customer
     * @return Customer
     */
    Customer modifyCustomer(Customer customer);

    /**
     * 移除客户
     *
     * @param customerId
     */
    void removeCustomer(long customerId);

    /**************************************门店*********************************************/

    /**
     * 分页查询门店
     * @param search
     * @param currentPage
     * @param limit
     * @return
     */
    Page<Store> pagingQueryStoreList(String search, Integer currentPage, Integer limit);

    /**
     * 查询所有门店
     *
     * @return List<Store>
     */
    List<Store> findAllStore();

    /**
     * 根据门店ID查询单个门店
     *
     * @param storeId
     * @return Store
     */
    Store findStoreByStoreId(long storeId);

    /**
     * 添加门店信息
     *
     * @param store
     * @return Store
     */
    Store addStore(Store store);

    /**
     * 修改门店信息
     *
     * @param store
     * @return Store
     */
    Store modifyStore(Store store);

    /**
     * 移除门店
     *
     * @param storeId
     */
    void removeStore(long storeId);

    /**************************************仓库*********************************************/

    /**
     * 分页查询仓库
     * @param search
     * @param currentPage
     * @param limit
     * @return
     */
    Page<Warehouse> pagingQueryWarehouseList(String search, Integer currentPage, Integer limit);

    /**
     * 查询所有仓库
     *
     * @return List<Warehouse>
     */
    List<Warehouse> findAllWarehouse();

    /**
     * 根据仓库ID查询单个仓库
     *
     * @param warehouseId
     * @return Warehouse
     */
    Warehouse findWarehouseByWarehouseId(long warehouseId);

    /**
     * 添加仓库信息
     *
     * @param warehouse
     * @return Warehouse
     */
    Warehouse addWarehouse(Warehouse warehouse);

    /**
     * 修改仓库信息
     *
     * @param warehouse
     * @return Warehouse
     */
    Warehouse modifyWarehouse(Warehouse warehouse);

    /**
     * 移除仓库
     *
     * @param warehouseId
     */
    void removeWarehouse(long warehouseId);

    /**************************************商品*********************************************/

    /**
     * 分页查询商品
     * @param typeId
     * @param search
     * @param currentPage
     * @param limit
     * @return
     */
    Page<Product> pagingQueryProductList(Integer typeId, String search, Integer currentPage, Integer limit);

    /**
     * 查询所有商品
     *
     * @return List<Product>
     */
    List<Product> findAllProduct();

    /**
     * 查询所有商品分类
     * @return
     */
    List<ProductType> findAllProductType();

    /**
     * 根据商品ID查询单个商品
     *
     * @param productId
     * @return Product
     */
    Product findProductByProductId(long productId);

    /**
     * 保存商品信息
     *
     * @param product
     * @return Product
     */
    Product addProduct(Product product);

    /**
     * 保存商品信息
     *
     * @param product
     * @return Product
     */
    Product modifyProduct(Product product);

    /**
     * 移除商品
     *
     * @param productId
     */
    boolean removeProduct(long productId);

}

File: ERP/erp-frontend/src/app/shared/service/info.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs/Observable';

@Injectable()
export class InfoService {

    constructor(private http: HttpClient) {
    }

    // 条件查询供应商
    pagingQuerySupplierList(selectedTypeId: string, search: string, pageNum: number, limit: number): Observable<any> {
        if (search === null) {
            search = '';
        }
        return this.http.get(`/api/erp/info/suppliers?typeId=${selectedTypeId}&search=${search}&currentPage=${pageNum}&limit=${limit}`);
    }

    // 获取所有供应商
    querySupplierList(): Observable<any> {
        return this.http.get(`/api/erp/info/suppliers/all`);
    }

    // 获取所有供应商分类
    querySupplierTypeList(): Observable<any> {
        return this.http.get(`/api/erp/info/supplier/types`);
    }

    // 新增供应商
    addSupplier(supplier: any): Observable<any> {
        return this.http.post(`/api/erp/info/supplier`, supplier);
    }

    // 修改供应商
    modifySupplier(supplier: any): Observable<any> {
        return this.http.put(`/api/erp/info/supplier`, supplier);
    }

    // 删除供应商
    deleteSupplier(supplierId: any): Observable<any> {
        return this.http.delete(`/api/erp/info/supplier?supplierId=${supplierId}`);
    }

    // 条件查询客户
    pagingQueryCustomerList(selectedTypeId: string, search: string, pageNum: number, limit: number): Observable<any> {
        if (search === null) {
            search = '';
        }
        return this.http.get(`/api/erp/info/customers?typeId=${selectedTypeId}&search=${search}&currentPage=${pageNum}&limit=${limit}`);
    }

    // 获取所有客户
    queryCustomerList(): Observable<any> {
        return this.http.get(`/api/erp/info/customers/all`);
    }

    // 获取所有客户分类
    queryCustomerTypeList(): Observable<any> {
        return this.http.get(`/api/erp/info/customer/types`);
    }

    // 新增客户
    addCustomer(customer: any): Observable<any> {
        return this.http.post(`/api/erp/info/customer`, customer);
    }

    // 修改客户
    modifyCustomer(customer: any): Observable<any> {
        return this.http.put(`/api/erp/info/customer`, customer);
    }

    // 删除客户
    deleteCustomer(customerId: any): Observable<any> {
        return this.http.delete(`/api/erp/info/customer?customerId=${customerId}`);
    }

    // 条件查询门店
    pagingQueryStoreList(search: string, pageNum: number, limit: number): Observable<any> {
        if (search === null) {
            search = '';
        }
        return this.http.get(`/api/erp/info/stores?&search=${search}&currentPage=${pageNum}&limit=${limit}`);
    }

    // 获取所有门店
    queryStoreList(): Observable<any> {
        return this.http.get(`/api/erp/info/stores/all`);
    }

    // 新增门店
    addStore(store: any): Observable<any> {
        return this.http.post(`/api/erp/info/store`, store);
    }

    // 修改门店
    modifyStore(store: any): Observable<any> {
        return this.http.put(`/api/erp/info/store`, store);
    }

    // 删除门店
    deleteStore(storeId: any): Observable<any> {
        return this.http.delete(`/api/erp/info/store?storeId=${storeId}`);
    }

    // 条件查询仓库
    pagingQueryWarehouseList(search: string, pageNum: number, limit: number): Observable<any> {
        if (search === null) {
            search = '';
        }
        return this.http.get(`/api/erp/info/warehouses?&search=${search}&currentPage=${pageNum}&limit=${limit}`);
    }

    // 获取所有仓库
    queryWarehouseList(): Observable<any> {
        return this.http.get(`/api/erp/info/warehouses/all`);
    }

    // 新增仓库
    addWarehouse(warehouse: any): Observable<any> {
        return this.http.post(`/api/erp/info/warehouse`, warehouse);
    }

    // 修改仓库
    modifyWarehouse(warehouse: any): Observable<any> {
        return this.http.put(`/api/erp/info/warehouse`, warehouse);
    }

    // 删除仓库
    deleteWarehouse(warehouseId: any): Observable<any> {
        return this.http.delete(`/api/erp/info/warehouse?warehouseId=${warehouseId}`);
    }

    // 条件查询产品
    pagingQueryProductList(selectedTypeId: string, search: string, pageNum: number, limit: number): Observable<any> {
        if (search === null) {
            search = '';
        }
        return this.http.get(`/api/erp/info/products?typeId=${selectedTypeId}&search=${search}&currentPage=${pageNum}&limit=${limit}`);
    }

    // 获取所有产品
    queryProductList(): Observable<any> {
        return this.http.get(`/api/erp/info/products/all`);
    }

    // 获取所有产品
    queryProductTypeList(): Observable<any> {
        return this.http.get(`/api/erp/info/product/types`);
    }

    // 新增产品
    addProduct(product: any): Observable<any> {
        return this.http.post(`/api/erp/info/product`, product);
    }

    // 修改产品
    modifyProduct(product: any): Observable<any> {
        return this.http.put(`/api/erp/info/product`, product);
    }

    // 删除产品
    deleteProduct(productId: any): Observable<any> {
        return this.http.delete(`/api/erp/info/product?productId=${productId}`);
    }

}
